app/Support/Modules/DatabaseActivator.php

```diff
@@ -0,0 +1,248 @@
+     *
+     * @return array
+     */
+    private function getModulesStatuses(): array
+    {
+        $this->scan();
+        $modules = \App\Models\Module::all();
+        $retVal = [];
+        foreach ($modules as $i) {
+            $retVal[$i->name] = $i->enabled;
+        }
+        return $retVal;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset(): void
+    {
+        (new \App\Models\Module())->truncate();
+        $this->modulesStatuses = [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enable(Module $module): void
+    {
+        $this->setActiveByName($module->getName(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function disable(Module $module): void
+    {
+        $this->setActiveByName($module->getName(), false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasStatus(Module $module, bool $status): bool
+    {
+        if (!isset($this->modulesStatuses[$module->getName()])) {
+            return $status === false;
+        }
+
+        return $this->modulesStatuses[$module->getName()] === $status;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setActive(Module $module, bool $active): void
+    {
+        $this->setActiveByName($module->getName(), $active);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setActiveByName(string $name, bool $status): void
+    {
+        $this->modulesStatuses[$name] = $status;
+        $this->writeDB($name, $status);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(Module $module): void
+    {
+        $module_name = $module->getName();
+        if (!isset($this->modulesStatuses[$module_name])) {
+            return;
+        }
+        unset($this->modulesStatuses[$module_name]);
+        $this->writeDB($module_name, false, 1);
+    }
+
+    /**
+     * Writes the activation statuses in a file, as json
+     *
+     * @param $name
+     * @param $status
+     * @param string $delete
+     */
+    private function writeDB($name, $status, $delete = ''): void
+    {
+        if (!empty($delete)) {
+            (new \App\Models\Module())->where([
+                'name' => $name,
+            ])->delete();
+        } else {
+            (new \App\Models\Module())->where([
+                'name' => $name,
+            ])->update([
+                'status' => $status,
+            ]);
+        }
+    }
+
+    /**
+     * Get & scan all modules.
+     *
+     * @return array
+     */
+    public function scan()
+    {
+        $paths = $this->getScanPaths();
+
+        $modules = [];
+
+        foreach ($paths as $key => $path) {
+            $manifests = $this->getFiles()->glob("{$path}/module.json");
+
+            is_array($manifests) || $manifests = [];
+
+            foreach ($manifests as $manifest) {
+                $name = Json::make($manifest)->get('name');
+
+                $modules[$name] = $this->createModule($name);
```

Comment:
Don't create the module, just return a list of the scanned modules. You need to create an array with the list of module names, and then use `whereNotIn()` on the model to trim the list to return only the modules which aren't in the database already. So if you get a module list of `X, Y, Z` and in the database is only `X`, whereNotIn will return `Y, Z`, and that's the list that should be returned from this call.

Reply:
Yeah, got it! Thanks

Reply:
Did it in another way. Can you please check?